### PR TITLE
220166_Align-Invoice-API-Models-PaymentRequest

### PR DESCRIPTION
**Pull request by fshehzad-defra**
220166, Align PaymentRequest model to spec

### CODE REVIEW

File: EST.MIT.Invoice.Api/Models/PaymentRequest.cs [hunk 1]
```diff
@@ -7,36 +7,57 @@ public class PaymentRequest
...
     [Range(2021, 2099, ErrorMessage = "Marketing Year must be between 2021 and 2099 ")]
     public int MarketingYear { get; init; }
+
     [JsonProperty("paymentRequestNumber")]
     public int PaymentRequestNumber { get; init; }
+
     [JsonProperty("agreementNumber")]
     public string AgreementNumber { get; init; } = default!;
-    [JsonProperty("contractNumber")]
-    public string ContractNumber { get; init; } = default!;
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api/Models/PaymentRequest.cs:23]: Can ContractNumber be commented out for now, not removed?

File: EST.MIT.Invoice.Api/Models/PaymentRequest.cs [hunk 2]
```diff
@@ -7,36 +7,57 @@ public class PaymentRequest
...
-    [JsonProperty("firmReferenceNumber")]
-    public long FirmReferenceNumber { get; init; } = default!;
-
-    [JsonProperty("singleBusinessIdentifier")]
-    public int SingleBusinessIdentifier { get; init; } = default!;
+    [JsonProperty("sbi")]
+    public int SBI { get; init; } = default!;
 
     [JsonProperty("vendorId")]
     public string VendorID { get; init; } = default!;
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api/Models/PaymentRequest.cs:63]: This should be 'Vendor' to match the PaymentHub property

File: EST.MIT.Invoice.Api/Models/PaymentRequest.cs [hunk 3]
```diff
@@ -7,36 +7,57 @@ public class PaymentRequest
...
+    public string Currency { get; init; } = default!;
+
+    [JsonProperty("description")]
+    public string Description { get; init; } = default!;
+
+    [JsonProperty("originalInvoiceNumber")]
+    public string OriginalInvoiceNumber { get; init; } = default!;
+
+    [JsonProperty("originalSettlementDate")]
+    public string OriginalSettlementDate { get; init; } = default!;
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api/Models/PaymentRequest.cs:32]: This should be DateTime.

File: EST.MIT.Invoice.Api/Models/PaymentRequestValidator.cs [hunk 4]
```diff
@@ -43,27 +41,27 @@ public PaymentRequestValidator(IReferenceDataApi referenceDataApi, ICachedRefere
...
             .WithMessage("SBI is not in valid range (105000000 .. 999999999)")
-            .When(invoiceHeader => string.IsNullOrWhiteSpace(invoiceHeader.VendorID) && invoiceHeader.FirmReferenceNumber == 0
-                    && invoiceHeader.SingleBusinessIdentifier != 0,
+            .When(invoiceHeader => string.IsNullOrWhiteSpace(invoiceHeader.VendorID) && invoiceHeader.FRN == 0
+                    && invoiceHeader.SBI != 0,
                 ApplyConditionTo.CurrentValidator);
 
         RuleFor(invoiceHeader => invoiceHeader.VendorID)
             .Length(6)
             .WithMessage("VendorID must be 6 characters")
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api/Models/PaymentRequestValidator.cs:63]: VendorID => Vendor. I think. They might come back with a display name change.

File: EST.MIT.Invoice.Api/Models/PaymentRequest.cs (hunk 2 above)
Comment (fshehzad-defra) [reply EST.MIT.Invoice.Api/Models/PaymentRequest.cs:63]: Done.

File: EST.MIT.Invoice.Api/Models/PaymentRequest.cs [hunk 5]
```diff
@@ -7,36 +7,57 @@ public class PaymentRequest
...
+
+    [JsonProperty("recoveryDate")]
+    public string RecoveryDate { get; init; } = default!;
```
[the remark this answers is not in the document]
Comment (fshehzad-defra) [reply EST.MIT.Invoice.Api/Models/PaymentRequest.cs:35]: Done

File: EST.MIT.Invoice.Api/Models/PaymentRequest.cs (hunk 3 above)
Comment (fshehzad-defra) [reply EST.MIT.Invoice.Api/Models/PaymentRequest.cs:32]: Done

File: EST.MIT.Invoice.Api/Models/PaymentRequestValidator.cs (hunk 4 above)
Comment (fshehzad-defra) [reply EST.MIT.Invoice.Api/Models/PaymentRequestValidator.cs:63]: Done

File: EST.MIT.Invoice.Api/Models/PaymentRequest.cs (hunk 1 above)
Comment (andygibbonsdefra) [reply EST.MIT.Invoice.Api/Models/PaymentRequest.cs:23]: ContractNumber is still removed rather than commented out?

**Review by andygibbonsdefra — APPROVED** [review]
Will leave the ContractNumber removed for now.

**Review by andygibbonsdefra — APPROVED** [review]

**Review by andygibbonsdefra — APPROVED** [review]

**Pull request merged**